utils/near-framed-read/src/framed_read.rs

```diff
@@ -0,0 +1,72 @@
+use crate::framed_impl::{FramedImpl, ReadFrame};
+
+use std::time::Duration;
+
+use futures_core::Stream;
+use tokio::io::AsyncRead;
+
+use crate::decoder::Decoder;
+use bytes::BytesMut;
+use futures_timer::Delay;
+use pin_project_lite::pin_project;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+pin_project! {
+    pub struct NearFramedRead<T, D> {
+        #[pin]
+        inner: FramedImpl<T, D, ReadFrame>,
+    }
+}
+
+impl<T, D> NearFramedRead<T, D>
```

Comment:
I don't think there's anything specific to near in the way this thing operates, a better name would be `ThrottledFrameRead`.

Reply:
That's a good idea.